File: src/personalities/ambiverts/terminal.ts
import { invoke } from "@tauri-apps/api/core";
import { logger } from "../../utility/logger";

/*
dusty::api::terminal::get_available_terminals,
dusty::api::terminal::open_terminal_at_path,
*/

const CMD_GET_AVAILABLE_TERMINALS = "get_available_terminals";
const CMD_OPEN_TERMINAL_AT_PATH = "open_terminal_at_path";

export async function getAvailableTerminalsIPC(): Promise<string[]> {
    try {
        let result = await invoke<string[]>(CMD_GET_AVAILABLE_TERMINALS);
        return result;
    } catch (error) {
        logger.error(`getAvailableTerminalsIPC error: ${error}`);
        return [];
    }
}

export async function openTerminalAtPathIPC(path: string, terminal: string): Promise<boolean> {
    try {
        await invoke(CMD_OPEN_TERMINAL_AT_PATH, { path, terminal });
        return true;
    } catch (error) {
        logger.error(`openTerminalAtPathIPC error: ${error}`);
        return false;
    }
}

File: src-tauri/src/dusty/api/system/terminal.rs
use std::path::PathBuf;

use crate::dusty::{logger::logger, system::terminal::{available_terminals, open_terminal_at}};


#[tauri::command]
pub fn get_available_terminals() ->Vec<String>{
    available_terminals()
}

#[tauri::command]
pub fn open_terminal_at_path(path:String,terminal:String)->Result<(),String>{
    let path_buf = PathBuf::from(path);
    open_terminal_at(&path_buf, &terminal).map_err(|err|{
        logger::error!("FAILED_OPEN_TERMINAL_AT",err.to_string());
        err.to_string()
    })?;
    Ok(())
}